specreduce/tracing.py

```diff
@@ -336,24 +352,41 @@ def __post_init__(self):
                     y_bins[i] = popt_i.mean_0.value
                     popt_tot = popt_i
 
+            if z_i.mask.all():  # all-masked bins when peak_method is 'centroid' or 'max'
+                warn_bins.append(i)
+
             elif self.peak_method == 'centroid':
                 z_i_cumsum = np.cumsum(z_i)
-                # find the interpolated index where the cumulative array reaches half the total
-                # cumulative values
+                # find the interpolated index where the cumulative array reaches
+                # half the total cumulative values
                 y_bins[i] = np.interp(z_i_cumsum[-1]/2., z_i_cumsum, ilum2)
 
+                # warning message for fully masked bin
+                # NOTE this reflects current behavior, should eventually be changed
+                # to set to nan by default (or zero fill / interpoate option once
+                # available) and warn accordingly.
+                warn_msg = f'largest bin index ({str(z_i_cumsum.shape[0])})'
+
             elif self.peak_method == 'max':
                 # TODO: implement smoothing with provided width
                 y_bins[i] = ilum2[z_i.argmax()]
 
-        # warn about fully-masked bins (which, currently, means any bin with a single masked value)
+                # warning message for fully masked bin
+                # NOTE: should eventually be changed to set to nan by default
+                # (or zero fill / interpoate option onceavailable) and warn.
+                warn_msg = 'zero'
+
+        # warn about fully-masked bins
         if len(warn_bins) > 0:
+
             # if there are a ton of bins, we don't want to print them all out
             if len(warn_bins) > 20:
                 warn_bins = warn_bins[0: 10] + ['...'] + [warn_bins[-1]]
+
             warnings.warn(f"All pixels in {'bins' if len(warn_bins) else 'bin'} "
                           f"{', '.join([str(x) for x in warn_bins])}"
-                          " are masked. Falling back on trace value from all-bin fit.")
+                          " are fully masked. Setting bin"
+                          f" peak{'s' if len(warn_bins) else ''} to {warn_msg}.")
```

Comment:
The variable name `warn_msg` was confusing to me until I got all the way down here... can we use something more verbose/explanatory?  Maybe `warn_adopted_value` or something similar?  

Or maybe just do the formatting based on `peak_method` directly, with something like:


```
f" peak{'s' if len(warn_bins) else ''} to {{'gaussian': 'nan', 'max': 'zero'}.get(self.peak_method)}.")
```